rs/registry/canister/src/mutations/do_set_subnet_operational_level/tests.rs

```diff
@@ -0,0 +1,346 @@
+}
+
+lazy_static! {
+    static ref GENERAL_PAYLOAD: SetSubnetOperationalLevelPayload =
+        SetSubnetOperationalLevelPayload {
+            subnet_id: Some(SubnetId::from(PrincipalId::new_user_test_id(777))),
+            operational_level: Some(operational_level::NORMAL),
+            ssh_readonly_access: Some(vec!["hello".to_string(), "world".to_string()]),
+            ssh_node_state_write_access: Some(GENERAL_SSH_NODE_STATE_WRITE_ACCESS.clone()),
+        };
+}
+
+#[test]
+fn test_validate_payload_no_subnet_ok() {
+    // Step 1: Prepare the world.
+    let mut registry = REGISTRY.clone();
+
+    // Step 2: Run code under test.
+
+    // This just changes the NodeRecord, not the SubnetRecord.
+    registry.do_set_subnet_operational_level(SetSubnetOperationalLevelPayload {
+        subnet_id: None,
+        operational_level: None,
+        ssh_readonly_access: None,
+
+        ssh_node_state_write_access: Some(vec![NodeSshAccess {
+            node_id: Some(*NODE_ID),
+            public_keys: Some(vec!["fake node state write public key".to_string()]),
+        }]),
+    });
+
+    // Step 3: Verify results.
+
+    // Step 3A.1: Verify SubnetRecord.
+    let new_subnet_record = registry.get_subnet_or_panic(*SUBNET_ID);
+    assert_eq!(new_subnet_record, ORIGINAL_SUBNET_RECORD.clone(),);
+
+    // Step 3A.2: Verify NodeRecord.
+    let new_node_record = registry.get_node_or_panic(*NODE_ID);
+    assert_eq!(
+        new_node_record,
+        NodeRecord {
+            ssh_node_state_write_access: vec!["fake node state write public key".to_string()],
+            ..ORIGINAL_NODE_RECORD.clone()
+        }
+    );
+}
+
+#[test]
+fn test_validate_payload_no_node_ok() {
+    // Step 1: Prepare the world.
+    let mut registry = REGISTRY.clone();
+
+    // Step 2: Run code under test.
+
+    // This just changes the NodeRecord, not the SubnetRecord.
+    registry.do_set_subnet_operational_level(SetSubnetOperationalLevelPayload {
+        subnet_id: Some(*SUBNET_ID),
+        operational_level: Some(operational_level::DOWN_FOR_REPAIRS),
+        ssh_readonly_access: Some(vec!["fake read-only public key".to_string()]),
+
+        ssh_node_state_write_access: None,
+    });
+
+    // Step 3: Verify results.
+
+    // Step 3A.1: Verify SubnetRecord.
+    let new_subnet_record = registry.get_subnet_or_panic(*SUBNET_ID);
+    assert_eq!(
+        new_subnet_record,
+        SubnetRecord {
+            is_halted: true,
+            ssh_readonly_access: vec!["fake read-only public key".to_string()],
+            ..ORIGINAL_SUBNET_RECORD.clone()
+        }
+    );
+
+    // Step 3A.2: Verify NodeRecord.
+    let new_node_record = registry.get_node_or_panic(*NODE_ID);
+    assert_eq!(new_node_record, ORIGINAL_NODE_RECORD.clone(),);
+}
+
+/// The situation tested here is degenerate, but not broken per se, and therefore, allowed.
+#[test]
+fn test_validate_payload_no_nothing_ok() {
```

Comment:
Also should not validate.

Reply:
Done.

I disagree though. The comment explains why I think this should be considered "ok".